main.tf

```diff
@@ -59,4 +59,14 @@ resource "aws_route53_record" "srv_records" {
   zone_id  = var.zone_id
 
   records = each.value.records
-}
\ No newline at end of file
+}
+
+resource "aws_route53_record" "ns_records" {
+  count   = length(var.ns_records) > 1 ? 1 : 0
+  name    = ""
```

Comment:
As we wanna add a custom nested NS record, `name` should define as a variable, perhaps `var.nested_ns_name`
consequently :

```
ses.ryte-test.com NS ns-1193.awsdns-21.org.
                                     ns-177.awsdns-22.com.
                                     ns-555.awsdns-05.net.
                                     ns-1880.awsdns-43.co.uk.
```

Reply:
code:
```hcl
module "ses_ryte_test" {
    zone_id = data.aws_route53_zone.zone.zone_id
    ttl  = 300
    nested_ns_name = "ses.ryte-test.com"
    ns_records = [
      "ns-1193.awsdns-21.org.",
      "ns-177.awsdns-22.com.",
      "ns-555.awsdns-05.net.",
      "ns-1880.awsdns-43.co.uk.",
    ]

    source = "https://github.com/ryte/INF-tf-route-53.git?ref=v0.2.2"
}
```
---
output:

```hcl
 # module.ses_ryte_test.aws_route53_record.ns_records[0] will be created
  + resource "aws_route53_record" "ns_records" {
      + allow_overwrite = (known after apply)
      + fqdn            = (known after apply)
      + id              = (known after apply)
      + name            = "ses.ryte-test.com"

      + records         = [
          + "ns-1193.awsdns-21.org.",
          + "ns-177.awsdns-22.com.",
          + "ns-1880.awsdns-43.co.uk.",
          + "ns-555.awsdns-05.net.",
        ]
      + ttl             = 300
      + type            = "NS"
      + zone_id         = "Z2MH9KE3TU4YTZ"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

Reply:
Well, if you [check here](https://github.com/onpage-org/INF-tf-global/blob/ca61bc0cddfff330269bc3f664bbf2ba7cc30968/stacks/r53/ryte_test_com/ryte_test.tf) is a module for the whole `ryte_test_com ` domain, so I would suggest to keep it like this and add a possibility to define named records, similar to `cname_records`, like this:
```
module "ryte_test_com" {
  zone_id = aws_route53_zone.ryte_test_com.id
  ttl     = 300

  ns_records = {
    1 = {
      name = "ses.ryte-test.com"
      records = [
        "ns-1193.awsdns-21.org.",
        "ns-177.awsdns-22.com.",
        "ns-555.awsdns-05.net.",
        "ns-1880.awsdns-43.co.uk."
      ]
    }
  }
....
}
```

Reply:
ok make sense 👍